llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/pooling.py

```diff
@@ -47,3 +50,13 @@ def mean_pooling(cls, array: np.ndarray) -> np.ndarray:
         if len(array.shape) == 2:
             return array.mean(axis=0)
         raise NotImplementedError(f"Unhandled shape {array.shape}.")
+
+    @classmethod
+    def last_pooling(
```

Comment:
We are using `Union[np.ndarray, torch.Tensor]` for the input types in our pooling methods (`cls_pooling` and `last_pooling`). To be consistent, I recommend adding overloaded method signatures for last_pooling, similar to what we've done with cls_pooling.

This would provide explicit information to the type checker about the different types of inputs that these methods can handle and what they return, depending on the input type. 

something like below should be sufficient:

```python
@classmethod
@overload
def last_pooling(cls, array: np.ndarray) -> np.ndarray:
    ...

@classmethod
@overload
def last_pooling(cls, array: "torch.Tensor") -> "torch.Tensor":  # type: ignore
    ...
```